code/modules/stock_market/events.dm

```diff
@@ -0,0 +1,223 @@
+#define TIME_MULTIPLIER 0.5 // so I can speed up/slow down shit
+
+/datum/stockEvent
+	var/name = "event"
+	var/next_phase = 0
+	var/datum/stock/company = null
+	var/current_title = "A company holding an pangalactic conference in the Seattle Conference Center, Seattle, Earth"
+	var/current_desc = "We will continue to monitor their stocks as the situation unfolds."
+	var/phase_id = 0
+	var/hidden = 0
+	var/finished = 0
+	var/last_change = 0
+
+/datum/stockEvent/process()
+	if (finished)
+		return
+	if (ticker.round_elapsed_ticks > next_phase)
```

Comment:
```world.time``` is the number of server ticks that have passed multiplied by ```tick_lag```
it's essentially a tick counter, it doesn't obey "time" as such, so it avoids midnight rollover.
you're confusing it with ```world.timeofday```